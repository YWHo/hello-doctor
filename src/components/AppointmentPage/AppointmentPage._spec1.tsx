import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { render, screen } from '@testing-library/react';
import AppointmentPage, { showTimeCards } from './AppointmentPage';
import { schedule as ScheduleType } from '../../state/state-interface';
import { DAY } from '../../shared/constants';

// Mock the Redux store
const mockStore = configureStore([]);
const store = mockStore({
  pendingAppointmentReducer: {
    selectedDate: new Date(0),
    selectedDayPart: DAY.MORNING,
    selectedTimeID: '1',
  },
  providerProfileReducer: {
    Degrees: [],
    Description: '',
    Id: '',
    Languages: [],
    Name: '',
    PictureName: '',
    PictureURL: '',
    Title: '',
  },
  timeSlotsReducer: {
    schedules: [
      {
        Id: 1,
        Title: 'Dr. Smith',
        AvailableSlots: { id1001: '9:00 AM', id1002: '9:00 AM' },
      },
      {
        Id: 2,
        Title: 'Dr. Johnson',
        AvailableSlots: { id1003: '9:30 AM', id1004: '10:30 AM' },
      },
    ],
  },
  uiShowReducer: {
    showingCalendar: false,
    showingProfile: false,
  },
});

// Mock the custom hooks
jest.mock('../../hooks/useActions');

describe('AppointmentPage Component', () => {
  it('renders AppointmentPage component correctly', () => {
    const mockUseActions = jest.fn();
    mockUseActions.mockReturnValue({});
    jest.mock('../../hooks/useActions', () => mockUseActions);

    render(
      <Provider store={store}>
        <AppointmentPage />
      </Provider>,
    );

    // Add more assertions based on your component's structure
    expect(screen.getByText('MonthContainer')).toBeInTheDocument();
    expect(screen.getByText('DateSelector')).toBeInTheDocument();
  });

  it('renders TimeCard components based on schedules', () => {
    const mockUseActions = jest.fn();
    mockUseActions.mockReturnValue({});
    jest.mock('../../hooks/useActions', () => mockUseActions);

    const sampleSchedules: ScheduleType[] = [
      {
        Id: 1,
        Title: 'Dr. Smith',
        AvailableSlots: { id1001: '9:00 AM', id1002: '9:00 AM' },
      },
      {
        Id: 2,
        Title: 'Specialist',
        Name: 'Dr. Johnson',
        AvailableSlots: { id1003: '9:30 AM', id1004: '10:30 AM' },
      },
    ];

    render(<>{showTimeCards({ schedules: sampleSchedules })}</>);

    // Add more assertions based on your TimeCard component's structure
    expect(screen.getByText('Dr. Smith')).toBeInTheDocument();
    expect(screen.getByText('9:00 AM')).toBeInTheDocument();
    expect(screen.getByText('Dr. Johnson')).toBeInTheDocument();
    expect(screen.getByText('10:00 AM')).toBeInTheDocument();
  });
});
