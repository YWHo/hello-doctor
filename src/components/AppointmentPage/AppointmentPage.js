import React from 'react';
import { connect } from 'react-redux';
import moment from 'moment';
import styled from 'styled-components';
import selectors from '../../state/selectors';
import NavBar from '../NavBar';
import MonthBar from '../MonthBar';
import FiveDayBar from '../FiveDaysBar';

const Container = styled.div`
  height: 100vh;
  min-width: 375px;
`;

export function AppointmentPage(props) {
  const { schedules = [] } = props;
  const nameList = Array.from(schedules).map((item, idx) => {
    const { AvailableSlots = {} } = item;
    console.log('item: ', item);
    const slots = Object.keys(AvailableSlots).map(key =>
      moment(AvailableSlots[key]).format('HH:mm')
    );
    return (
      <div key={`app_${idx}`}>
        <div>Name: {item.Name}</div>
        <div>Title: {item.Title}</div>
        <div>Available Slots: {slots.join(' | ')}</div>
        <br />
      </div>
    );
  });

  return (
    <Container>
      <NavBar />
      <MonthBar />
      <FiveDayBar />
      {nameList}
    </Container>
  );
}

const mapStateToProps = state => {
  return {
    schedules: selectors.getSchedules(state)
  };
};

function mapDispatchToProps(dispatch) {
  return {};
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AppointmentPage);
