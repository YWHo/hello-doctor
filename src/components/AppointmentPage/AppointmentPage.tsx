import React from 'react';
import styled from 'styled-components';
import DateSelector from '../DateSelector';
import NavBar from '../NavBar';
import MonthBar from '../MonthBar';
import MonthCalendar from '../MonthCalendar';
import MonthContainer from '../MonthContainer';
import TimeCard from '../TimeCard';
import DoctorProfile from '../DoctorProfile';
import useTypedSelector from '../../hooks/useTypedSelector';
import { schedule as scheduleType } from '../../state/state-interface';

const Container = styled.div`
  background-color: #f5f5f5;
  height: 100vh;
  min-width: 300px;
`;

export default function AppointmentPage() {
  const { schedules = [] } = useTypedSelector(
    (state) => state.timeSlotsReducer,
  );
  return (
    <Container>
      <NavBar />
      <MonthContainer>
        <MonthBar />
        <MonthCalendar />
      </MonthContainer>
      <DateSelector />
      {showTimeCards({ schedules })}
      <DoctorProfile />
    </Container>
  );
}

interface showTimeCardsProps {
  schedules: scheduleType[];
}

function showTimeCards(props: showTimeCardsProps) {
  const { schedules = [] } = props;
  return Array.from(schedules).map((scheduleObj, idx) => {
    return <TimeCard meetSchedule={scheduleObj} key={`tc_${idx}`} />;
  });
}
