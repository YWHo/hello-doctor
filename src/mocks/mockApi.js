import * as mockData from './mockData';
import * as mockData2 from './mockData2';

export function mockFetchSchedule(date) {
  const scheduleDict2 = mockData2.getScheduleDict2(date);
  let data;

  if (process.env.NODE_ENV === 'test') {
    data = mockData.scheduleDict;
  } else {
    data = scheduleDict2;
  }

  if (data[date]) {
    return Promise.resolve(data[date]);
  } else {
    return Promise.resolve([]);
  }
}

export function mockFetchProvider(id) {
  const { providers } = mockData;
  const { providers2 } = mockData2;
  const data = { ...providers, ...providers2 };
  console.log('provider data:\n', data);

  if (data[id]) {
    return Promise.resolve(data[id]);
  } else {
    return Promise.resolve({});
  }
}
