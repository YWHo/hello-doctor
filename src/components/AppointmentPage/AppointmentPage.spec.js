import React from 'react';
import { AppointmentPage } from './AppointmentPage';
import { mockFetchSchedule } from '../../mocks/mockApi';
import renderer from 'react-test-renderer';

describe('AppointmentPage', () => {
  it('render correctly', () => {
    return mockFetchSchedule('2019-06-13').then(schedules => {
      const component = renderer.create(
        <AppointmentPage schedules={schedules} />
      );
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
