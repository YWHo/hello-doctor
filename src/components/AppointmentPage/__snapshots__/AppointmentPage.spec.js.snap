// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentPage render correctly 1`] = `
<styled.div>
  <NavBar />
  <Connect(MonthBar) />
  <Connect(DateSelector) />
  <Connect(TimeCard)
    key="tc_0"
    meetSchedule={
      Object {
        "AvailableSlots": Object {
          "0d2e54d4-9854-447a-a9b0-39187b77b789": "2019-06-16T11:30:00",
          "10db464a-6b1a-47f0-bcf0-16a07909693d": "2019-06-16T14:15:00",
          "12cbdc25-9fc7-4f4e-85f0-04d9f5e84913": "2019-06-16T13:00:00",
          "133d8e87-6c55-44c4-aed7-f19c835973b6": "2019-06-16T11:00:00",
          "15465fc7-56f6-4e99-9d49-e26fb4773927": "2019-06-16T08:45:00",
          "2805e2bd-f884-4f72-b3ed-e71424bab0b0": "2019-06-16T16:45:00",
          "2d9031d4-1ed0-48c9-8138-681c6410eafe": "2019-06-16T17:15:00",
          "3069c8fd-f128-4ca0-b6f7-38c06d31e41c": "2019-06-16T18:15:00",
          "3a9319a6-2099-48fd-8e07-a33e883f4d7b": "2019-06-16T20:00:00",
          "3ea1c27a-b0e5-4e25-af5e-8e0f9ad159a2": "2019-06-16T10:00:00",
          "3faa10e0-f253-4eda-ac1e-5038c5566b19": "2019-06-16T19:15:00",
          "426c2477-5cda-44a1-9354-14c33c5e90a7": "2019-06-16T10:15:00",
          "459423e4-306a-4b52-b4b1-38516850a131": "2019-06-16T15:30:00",
          "464fa55d-47dd-4cd4-826f-d02ccbc09211": "2019-06-16T17:45:00",
          "523d9d66-32a9-4f94-90fb-6fb412545bcd": "2019-06-16T09:30:00",
          "546ad6da-1fe7-4081-a820-3c9e838aa453": "2019-06-16T09:45:00",
          "5687a4cc-1166-4528-a836-0fe2fd432be8": "2019-06-16T19:30:00",
          "59066218-41ac-4966-b701-599d4e0de8e4": "2019-06-16T12:15:00",
          "59d106a1-9d5c-4f27-acc1-a48102b05f90": "2019-06-16T16:30:00",
          "5a6f2f7b-afe6-47db-b464-f3fd3ee60a31": "2019-06-16T15:15:00",
          "5e3d85ec-62f2-4af0-b330-67fa4d1d4464": "2019-06-16T19:45:00",
          "68123fb0-14f2-4a5a-8182-277ae359f3d9": "2019-06-16T14:45:00",
          "6b721dd9-1ec6-4961-adeb-507bfd91426f": "2019-06-16T19:00:00",
          "852c96f5-cd20-4ebd-96de-50cc3f66ac64": "2019-06-16T18:30:00",
          "8b1d66d0-109a-428c-8f5c-3f9e9bf109bf": "2019-06-16T18:45:00",
          "95dee031-6cb7-4644-b32e-9714c6b3e59d": "2019-06-16T12:45:00",
          "95e43cbd-497f-4060-b83e-018e979d455b": "2019-06-16T15:45:00",
          "9be3c9db-450e-4dbd-b6f7-b584b5b1c8aa": "2019-06-16T09:15:00",
          "a2341049-0eb3-4958-b735-2ab2222963cc": "2019-06-16T13:30:00",
          "a8e4c382-e552-4a9d-b79f-008f73ee4edc": "2019-06-16T13:15:00",
          "aad89b10-872e-44bf-a838-3bae0cc9f2f8": "2019-06-16T14:00:00",
          "b1494e9d-42aa-4d31-bd96-a5c5de557c4b": "2019-06-16T08:30:00",
          "b433b622-73ff-4062-aee4-4c6310e83a34": "2019-06-16T15:00:00",
          "bd00a08f-3493-48c7-8f4f-a4a2dfe84f0a": "2019-06-16T10:30:00",
          "bd0d5497-c62b-47aa-8493-5aaf335aa3b6": "2019-06-16T10:45:00",
          "c6c5a1af-892c-4ece-b669-a13657b829a2": "2019-06-16T17:00:00",
          "c8f3f7fb-315d-4f13-a043-cba94c7c643b": "2019-06-16T12:00:00",
          "cd56978a-2f58-4668-825a-d30e24a40ee3": "2019-06-16T11:45:00",
          "d086c8dc-e70a-49da-9bd3-22cf172be6f8": "2019-06-16T18:00:00",
          "d54c16cf-44dc-4352-ad68-6380aeba9f2a": "2019-06-16T08:15:00",
          "d6cbca46-e010-457d-8a7e-c6b3a28d729b": "2019-06-16T08:00:00",
          "d6effe24-93a4-4e5d-8474-0c64059b3d8f": "2019-06-16T09:00:00",
          "d740d953-5410-4540-88bd-24cc84469ba8": "2019-06-16T17:30:00",
          "d804fcd8-a49b-4eb9-8938-4523f57f2410": "2019-06-16T14:30:00",
          "d8b7b0d1-1919-4224-8278-ba2d8482928c": "2019-06-16T11:15:00",
          "e7c5cb9a-51df-452e-aa30-5becaf5683ca": "2019-06-16T16:15:00",
          "eae46c56-85b2-4e4a-9d89-2527a43c6ead": "2019-06-16T16:00:00",
          "f4e5eeaa-302a-4186-b5ce-0cd5d687005e": "2019-06-16T12:30:00",
          "fb3cc3d0-b4ce-4599-bc9e-d4d50a45f1a0": "2019-06-16T13:45:00",
        },
        "Id": "e538c28d-52df-0654-800f-6145909aca28",
        "Name": "Dr. John Smith",
        "PictureURL": "/api/Picture/e538c28d-52df-0654-800f-6145909aca28",
        "Title": "General Practitioner",
      }
    }
  />
  <Connect(TimeCard)
    key="tc_1"
    meetSchedule={
      Object {
        "AvailableSlots": Object {
          "0d371459-1129-4437-81f1-e7b7af7376da": "2019-06-16T11:00:00",
          "186cf9d6-aeea-42a0-bcb4-42798f4d1b27": "2019-06-16T08:30:00",
          "2a306785-3093-4e47-829d-5342c66b5c51": "2019-06-16T09:30:00",
          "332c5155-bc01-4df1-8a2c-76d0849afb16": "2019-06-16T10:00:00",
          "357ea510-a2cb-41b6-a67a-d18a02a12260": "2019-06-16T09:15:00",
          "543fec11-2d4a-4328-8ef0-2d6c50bcbf16": "2019-06-16T11:45:00",
          "5912c71b-fab4-4390-aed4-31609632d8fc": "2019-06-16T11:30:00",
          "655e17e7-487c-4680-9aa1-41a610095d8a": "2019-06-16T09:00:00",
          "744022e0-54fb-41b5-b042-b4f8810d7e38": "2019-06-16T12:00:00",
          "7a2d32c2-5f72-4b82-b500-0bcc3fc18c1a": "2019-06-16T08:15:00",
          "9379caf0-596c-4e34-80da-48904c001c54": "2019-06-16T09:45:00",
          "9e7bbebf-92b7-4f1b-a6fc-35a72778cd77": "2019-06-16T08:00:00",
          "a053fc80-c5ea-41b6-823a-85e70dee993d": "2019-06-16T11:15:00",
          "a65fe281-708b-47ad-83aa-644afc76d1fb": "2019-06-16T10:30:00",
          "c288a36e-2b41-4ecc-8d34-7d0ff543622c": "2019-06-16T10:45:00",
          "cb93ca92-c95d-44cc-9764-5e371490ee6c": "2019-06-16T10:15:00",
          "ee66c958-73d6-48a2-8645-1bbf549ad51f": "2019-06-16T08:45:00",
        },
        "Id": "40abb954-2f57-4106-61ec-ddf2acfbf8ed",
        "Name": "Dr. Mary Aniston",
        "PictureURL": "/api/Picture/40abb954-2f57-4106-61ec-ddf2acfbf8ed",
        "Title": "General Practitioner",
      }
    }
  />
  <Connect(TimeCard)
    key="tc_2"
    meetSchedule={
      Object {
        "AvailableSlots": Object {
          "0656cb27-0be3-4d92-81d7-f752692e57cd": "2019-06-16T18:15:00",
          "12dbd846-fc64-4c2a-82fa-7d52f4476309": "2019-06-16T13:30:00",
          "197ba5db-7e8d-4b1b-9685-55a2ba91d8a8": "2019-06-16T14:00:00",
          "1b166742-6ced-4977-8386-168472c1ce6e": "2019-06-16T12:45:00",
          "1da6ea05-1e61-4e28-81fd-75d83a2f15fd": "2019-06-16T14:45:00",
          "2fc0c30a-6581-4ff1-89a2-3f5d5af672c9": "2019-06-16T13:45:00",
          "33528206-8703-4483-8c0c-fe7cf8471906": "2019-06-16T19:15:00",
          "3f0f3d3b-4bc0-4b6e-ae1a-cdb510719f6c": "2019-06-16T16:00:00",
          "44c145a5-aed1-4eca-81f4-4908a766a54f": "2019-06-16T18:30:00",
          "45f4bf0b-8908-41bb-9c51-daee264ab963": "2019-06-16T14:15:00",
          "499a86ab-f684-4aaa-94f6-749cdeea9ae6": "2019-06-16T13:00:00",
          "524a3bde-f131-4453-99e0-1ef45df39d38": "2019-06-16T17:45:00",
          "53935cdb-d892-4648-a328-08cc3a43e849": "2019-06-16T19:30:00",
          "5877c117-1688-45d4-8feb-8b5b2ac743bd": "2019-06-16T16:15:00",
          "5ca8fc83-8f5e-40c8-be33-6f34aa8e7f61": "2019-06-16T12:30:00",
          "67119487-e0f7-44f6-9104-94c2def232ad": "2019-06-16T15:30:00",
          "70cf2dc8-538e-48cf-a0e6-5c8ecc0174fb": "2019-06-16T13:15:00",
          "87e5128b-d134-4d67-aca2-eef56a800adc": "2019-06-16T17:00:00",
          "8b00d93e-2a41-42d0-b164-ab19e0002a1c": "2019-06-16T20:00:00",
          "9ba90e2c-885d-4f91-b811-a50de27bf346": "2019-06-16T19:00:00",
          "9cb5b97c-203f-42be-a9e7-b8fda99c63f6": "2019-06-16T18:45:00",
          "a01fe071-d5f1-477e-aed4-1d76f51ee02b": "2019-06-16T19:45:00",
          "aa0fb9f8-6c67-4ab3-87ff-76c1e1d708d2": "2019-06-16T15:00:00",
          "ad090304-77ff-40c1-8574-c37dd306dd1f": "2019-06-16T16:45:00",
          "c43d5ad8-3100-4c27-a3e7-8bc75b0505f8": "2019-06-16T15:15:00",
          "c670ffdd-345a-4b53-b767-87ec5c0eefba": "2019-06-16T18:00:00",
          "cd485d07-07a0-4050-b8bf-a5f4897d7999": "2019-06-16T15:45:00",
          "d3513ffb-e699-4f31-9515-df1b90782d42": "2019-06-16T17:30:00",
          "d987cba9-c981-48df-aa44-175dbea979b4": "2019-06-16T12:00:00",
          "dc80b92c-3e24-464e-b1a7-edb2ef676c76": "2019-06-16T16:30:00",
          "df111c63-22c2-4acf-8efc-8148e3dc0ac9": "2019-06-16T12:15:00",
          "f740132c-a44e-4134-8aed-5a8b70c15bf4": "2019-06-16T17:15:00",
          "fbce5dce-768e-450d-a30a-9c5d63892fa7": "2019-06-16T14:30:00",
        },
        "Id": "8dfe2600-52d5-3b80-129a-45d796fbaefd",
        "Name": "Dr. Sarah Lee Mance",
        "PictureURL": "/api/Picture/8dfe2600-52d5-3b80-129a-45d796fbaefd",
        "Title": "General Practitioner",
      }
    }
  />
</styled.div>
`;
